src/wasm/WasmModule.cpp

```diff
@@ -485,74 +567,74 @@ uint32_t WasmModule::createMemoryGuardRegion(uint32_t wasmOffset)
 void WasmModule::queuePthreadCall(threads::PthreadCall call)
 {
     queuedPthreadCalls.emplace_back(call);
+    isQueuedPthreadCalls.store(true);
```

Comment:
Again, we may want to pass `std::memory_order_release` and `std::memory_order_acquire_release` for these atomic operations (`store` and `fetch_add`).